### DISCUSSION TITLE
The middleware.ts file is not being recognized by the app.

### DISCUSSION THREAD

**Discussion by izanoth** [vercel/next.js#74215] [Help]
### Summary

we have an app with the following structure (or something like so)

app/
app/(admin)
app/(admin)/page.ts
app/(admin)/layout.ts
app/(main)/
app/(main)/page.ts
app/(main)/page.ts
app/globals.css
app/midleware.ts

the intend is to preserve admin page to be accessed by unauthorizated users.
for that reason, we have middleware.ts at the top level directory, as it must be according the official documentation 
https://nextjs.org/docs/pages/building-your-application/routing/middleware

below is the our `middleware.ts` file
```
import { NextResponse } from "next/server";
import type { NextRequest } from "next/server";

export function middleware(req: NextRequest) {
  console.log("Middleware executado para:", req.nextUrl.pathname); // Debug
  
  const token = req.cookies.get("authToken") || req.headers.get("authorization");

  if (req.nextUrl.pathname.startsWith("/admin") && token !== "valid-token") {
    return NextResponse.redirect(new URL("/", req.url));
  }

  return NextResponse.next();
}

export const config = {
  matcher: ["/admin/:path*"],
};
```

The debugging message is not displayed on console, that is, the file is not being read by.
We move this file for all sort of paths inside the app.

Thanks for any helps.

**Comment by icyJoseph** [comment]
> Use the file middleware.ts (or .js) in the root of your project to define Middleware. For example, at the same level as app or pages, or inside src if applicable.

That means that you should have:

```console
app/
app/(admin)
app/(admin)/page.ts
app/(admin)/layout.ts
app/(main)/
app/(main)/page.ts
app/(main)/page.ts
app/globals.css
middleware.ts 
```

With `middleware.ts` as a sibling, not a child, to the `app` folder.

**Reply by izanoth** [reply]
Thanks @icyJoseph , strangly it worked :)

**Reply by icyJoseph** [reply]
Awesome, please don't forget to mark as answered!
